RevenueCatUI/CustomerCenter/Utilities/CustomerCenterStoreKitUtilities.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CustomerCenterStoreKitUtilities.swift
+//
+//  Created by Will Taylor on 12/18/24.
+
+import Foundation
+import StoreKit
+
+import RevenueCat
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, visionOS 1.0, *)
+class CustomerCenterStoreKitUtilities: CustomerCenterStoreKitUtilitiesType {
+
+    func renewalPriceFromRenewalInfo(for product: StoreProduct) async -> (price: Decimal, currencyCode: String)? {
+
+        #if compiler(>=6.0)
+        guard let renewalInfo = await renewalInfo(for: product) else { return nil }
+        guard let renewalPrice = renewalInfo.renewalPrice else { return nil }
+        guard let currencyCode = currencyCode(fromRenewalInfo: renewalInfo) else { return nil }
+
+        return (renewalPrice, currencyCode)
+        #else
+        return nil
+        #endif
+    }
+
+    private func currencyCode(
+        fromRenewalInfo renewalInfo: Product.SubscriptionInfo.RenewalInfo,
+        locale: Locale = Locale.current
+    ) -> String? {
+        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, watchOSApplicationExtension 9.0, *) {
+
+            // renewalInfo.currency was introduced in iOS 18.0 and backdeployed through iOS 16.0
+            // However, Xcode versions <15.0 don't have the symbols, so we need to check the compiler version
+            // to make sure that this is being built with an Xcode version >=15.0.
+            #if compiler(>=6.0)
+            guard let currency = renewalInfo.currency else { return nil }
+            if currency.isISOCurrency {
```

Comment:
interesting, what's the other option if it's not ISO? just curious

Reply:
I'm honestly not sure, but `NumberFormatter.currencyCode` depends on the currency code being a ISO 4217 currency code, so I figured it was worth a check here